lib/config/systemd.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"io"
+	"os"
+	"text/template"
+
+	"github.com/gravitational/trace"
+)
+
+const (
+	// DefaultEnvironmentFile is the default path to the env file
+	DefaultEnvironmentFile = "/etc/default/teleport"
+	// DefaultPIDFile is the default path to the PID file
+	DefaultPIDFile = "/run/teleport.pid"
+	// DefaultFileDescriptorLimit is the default max number of open file descriptors
+	DefaultFileDescriptorLimit = 8192
+)
+
+// systemdUnitFileTemplate is the systemd unit file configuration template.
+var systemdUnitFileTemplate = template.Must(template.New("").Parse(`[Unit]
+Description=Teleport Service
+After=network.target
+
+[Service]
+Type=simple
+Restart=on-failure
+EnvironmentFile=-{{ .EnvironmentFile }}
+ExecStart={{ .TeleportInstallationFile }} start --pid-file={{ .PIDFile }}
+ExecReload=/bin/kill -HUP $MAINPID
+PIDFile={{ .PIDFile }}
+LimitNOFILE={{ .FileDescriptorLimit }}
+
+[Install]
+WantedBy=multi-user.target`))
+
+// SystemdSampleFlags specifies configuration parameters for a systemd unit file.
+type SystemdSampleFlags struct {
```

Comment:
We aren't really generating a "sample" systemd file, we're generating a real one that we expect people to use.

Should we rename to `SystemdFlags` instead?